lib/internal/worker.js

```diff
@@ -527,6 +548,97 @@ function eventLoopUtilization(util1, util2) {
   );
 }
 
+function setConnectionsListener(fn) {
+  if (isMainThread && !mainPortWasSetup) {
+    setupMainPort();
+    mainPortWasSetup = true;
+  }
+
+  if (fn == null) {
+    connectionsListener = null;
+    return;
+  }
+
+  validateFunction(fn, 'fn');
+  connectionsListener = fn;
+}
+
+async function processConnectionRequest(request) {
+  const status = new Int32Array(request.memory);
+
+  try {
+    const result = await connectionsListener?.(request.sender, request.port, request.data);
+    AtomicsStore(status, WORKER_MESSAGING_RESULT_INDEX, result === true ? 0 : 1);
+  } catch (e) {
+    debug('connections listener rejected', e);
+    AtomicsStore(status, WORKER_MESSAGING_RESULT_INDEX, 2);
+  } finally {
+    AtomicsNotify(status, WORKER_MESSAGING_STATUS_INDEX, 1);
+  }
+}
+
+async function connect(target, data, timeout) {
+  if (typeof data === 'number' && typeof timeout === 'undefined') {
+    timeout = data;
+    data = undefined;
+  }
+
+  if (typeof timeout !== 'undefined') {
+    validateNumber(timeout, 'timeout', 0);
+  }
+
+  if (target === threadId) {
+    throw new ERR_WORKER_SAME_THREAD();
+  }
+
+  // Create a shared array to exchange the status and the result
+  const memory = new SharedArrayBuffer(WORKER_MESSAGING_SHARED_DATA);
+  const status = new Int32Array(memory);
+  const promise = AtomicsWaitAsync(status, WORKER_MESSAGING_STATUS_INDEX, 0, timeout).value;
+
+  // Create the channel and send it to the other thread
+  const { port1, port2 } = new MessageChannel();
+
+  // Make sure the event loop does not exist before this is completed
+  port1.ref();
```

Comment:
The `ref()` and `unref()` a few lines below are on a sync call path. Is it really needed?
And if it is, why is it not needed across the async `await promise` below?